lib/puppet/provider/package/rpm.rb

```diff
@@ -313,7 +314,13 @@ def insync?(is)
     return false if [:purged, :absent].include?(is)
     return false if is.include?(self.class::MULTIVERSION_SEPARATOR) && !@resource[:install_only]
 
-    should = resource[:ensure]
+    should = @resource[:ensure]
+    if should =~ /[<>=~]/
+      should_version = Puppet::Util::Package::Version::Range.parse(should, Version::Rpm)
+      is_version = Version::Rpm.parse(is)
+      return should_version.include?(is_version)
+    end
```

Comment:
begin
       should_version = Puppet::Util::Package::Version::Range.parse(should, Version::Rpm)
    rescue Puppet::Util::Package::Version::Range::ValidationFailure, Puppet::Util::Package::Version::Range::Rpm::ValidationFailure
      Puppet.debug("Cannot parse #{should} as a RPM version range")
      return false
    end

    begin
      is_version = Version::Rpm.parse(is)
    rescue Version::Rpm::ValidationFailure
      Puppet.debug("Cannot parse #{is} as a rpm version")
      return false
    end
should_range.include?(is_version)